Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -234,23 +231,68 @@ private async Task<T> TryReadRemoteModelRevision(string modelRevisionExternalId,
             return null;
         }
 
-        /// <inheritdoc/>
-        public async Task<T> GetModelRevision(string modelRevisionExternalId)
+        /// <summary>
+        /// If model revision is not found in the local state, this method will try to read it from CDF, download the file and extract the model information.
+        /// If the model revision is already in the local state and has not been changed, it will return the existing state.
+        /// </summary>
+        /// <param name="modelRevisionExternalId">Model revision External ID</param>
+        /// <param name="token">Cancellation token</param>
+        private async Task<T> GetOrAddModelRevisionImpl(string modelRevisionExternalId, CancellationToken token = default)
         {
-            var modelVersions = _state.Values
-                .Where(s => s.ExternalId == modelRevisionExternalId
-                    && s.IsExtracted
-                    && !string.IsNullOrEmpty(s.FilePath))
-                .OrderByDescending(s => s.CreatedTime);
+            var modelRevision = await TryReadRemoteModelRevision(modelRevisionExternalId, token).ConfigureAwait(false);
+
+            if (modelRevision == null)
+            {
+                _logger.LogError("Model revision {ModelRevisionExternalId} not found in CDF", modelRevisionExternalId);
+                return null;
+            }
 
-            var model = modelVersions.FirstOrDefault();
+            _state.TryGetValue(modelRevision.Id.ToString(), out var modelState);
```

Comment:
Why the `Id.ToString()`? This smells a little to me.

Reply:
I agree this can be improved, but it's been stored like this (key as strings) even before this PR, so I don't want to pull more changes in (I can create a ticket ,maybe?)